windows fileshare support from within jupyter

RGD2 opened the pull request:
This will remain in the 'cifs' fork so as not to clutter `master`, and because it's not everyone's cup of tea.

I've added some guidance to the readme to make it less scary for newcomers to use.

RGD2 committed:
adds cifs support for accessing shared data

makes a mount point s visible within jupyter with a connection to a cifs server \\server\share with u/p shareusername/sharepassword
You'll probably want to make your own private branch of this, if you use
it, and change the credentials or use a credentials file you don't
commit.

RGD2 committed:
adds support for accessing a windows fileshare

with documentation changes for clarity.

vsoch commented on Singularity:
```diff
@@ -3,6 +3,8 @@ From: continuumio/anaconda3
 %runscript
 
+     echo "Mounting shared drive..."
+     /sbin/mount.cifs //server/share /opt/notebooks/share -o user=shareusername,password=sharepassword
```
Could this be passed in to the container from the host running it?

E.g.,

```bash
shareuser=${1}
sharepass=${2}

...
/sbin/mount.cifs //server/share /opt/notebooks/share -o user=${shareuser},password=${sharepass}
```

of course you would want to do a better job of checking that both are defined and then spitting out an error if not. Generally, I would never suggest or store any kind of credential in a git repo, so I wouldn't want to suggest for the user to do it. Thoughts?

RGD2 replied:
Good idea, I'll have a go at it.

RGD2 replied:
Got stuck quickly -- how does one pass parameters to the runscript when using a sandbox and not an executable image?
`sudo singularity run --writable jupyter-box server share user pw` doesn't work

vsoch replied:
It should be exactly the same! If you want to post what you have so far I can help tomorrow, going to bed soon :)

RGD2 replied:
regardless of what arguments I add to that line, I get: 
```bash
/.singularity.d/runscript: 4: /.singularity.d/runscript: server: not found
/.singularity.d/runscript: 5: /.singularity.d/runscript: share: not found
/.singularity.d/runscript: 6: /.singularity.d/runscript: shareuser: not found
/.singularity.d/runscript: 7: /.singularity.d/runscript: sharepw: not found
Mounting shared drive /// with username
mount.cifs: bad UNC (///)
Starting notebook...
```

RGD2 replied:
```diff
diff --git a/Singularity b/Singularity
index 694fa50..36596d5 100644
--- a/Singularity
+++ b/Singularity
@@ -3,8 +3,12 @@ From: continuumio/anaconda3

 %runscript

-     echo "Mounting shared drive..."
-     /sbin/mount.cifs //server/share /opt/notebooks/share -o user=shareusername,password=sharepassword
+     server = ${1}
+     share = ${2}
+     shareuser = ${3}
+     sharepw = ${4}
+     echo "Mounting shared drive //${server}/${share} with username ${shareuser}"
+     /sbin/mount.cifs //${server}/${share} /opt/notebooks/share -o user=${shareuser},password=${sharepw}
      echo "Starting notebook..."
      echo "Open browser to localhost:8888"
      exec /opt/conda/bin/jupyter notebook --notebook-dir=/opt/notebooks --ip='*' --port=8888 --no-browser --allow-root
```

vsoch replied:
You can’t have spaces with bash variables, that’s the issue.

vsoch replied:
Also it’s best practice to put them in double quotes.

RGD2 committed:
adds support for mounting //server/sharename

- will be visible within jupyter as ./share/
- does not require any file editing to use
- no credentials stored in repo

RGD2 commented:
Whoops. I knew that but had forgotten. Swapping and changing between too many languages is bad this way. 

Should be good now.

vsoch reviewed (approved):
hey it happens to the best of us! This looks good so I'm happy to merge, and I'll update the main README with a link to it. Thank you @RGD2 !
